zeroex/order.go

```diff
@@ -43,6 +45,21 @@ type SignedOrder struct {
 	Signature             []byte         `json:"signature"`
 }
 
+// SignatureType represents the type of 0x signature encountered
+type SignatureType uint8
+
+// SignatureType values
+const (
+	IllegalSignature SignatureType = iota
+	InvalidSignature
+	EIP712Signature
+	EthSignSignature
+	WalletSignature
+	ValidatorSignature
+	PreSignedSignature
+	NSignatureTypesSignature
```

Comment:
I had to suffix these values with `Signature` because there was already a collision between the `Invalid` value in this enum, and the `Invalid` value in the `OrderStatus` enum.